cpp/tests/utilities/identify_stream_usage.cpp

```diff
@@ -66,6 +67,34 @@ rmm::cuda_stream_view const get_default_stream()
 }  // namespace test
 #endif
 
+#ifdef STREAM_MODE_TESTING
+namespace detail {
+
+/**
+ * @brief Implementation of `cuda_stream_pool` that always returns the
+ * `cudf::test::get_default_stream()`
+ */
+class test_cuda_stream_pool : public cuda_stream_pool {
+ public:
+  rmm::cuda_stream_view get_stream() override { return cudf::test::get_default_stream(); }
+  rmm::cuda_stream_view get_stream(stream_id_type stream_id) override
+  {
+    return cudf::test::get_default_stream();
+  }
+
+  std::vector<rmm::cuda_stream_view> get_streams(std::size_t count) override
+  {
+    return std::vector<rmm::cuda_stream_view>(count, cudf::test::get_default_stream());
+  }
+
+  std::size_t get_stream_pool_size() const override { return 1UL; }
```

Comment:
Hmm I'm not sure how exactly this function is meant to be used. Is it coupled with calls to `get_streams`? If so, then really the size is infinite since we can request an arbitrary size subset of the pool and get that many copies of the same stream. Is returning 1 here consistent with those semantics?

Reply:
The comment above the `get_streams` declaration in `include/cudf/detail/utilities/stream_pool.hpp` suggests that `get_streams(int n)` returns $n$ unique streams from the pool if $n \leq$ `get_stream_pool_size()`. I returned 1 here since the test stream pool has one unique stream.

Reply:
I see, yes that makes sense. I made a small suggestion there to make the uniqueness explicit, otherwise LGTM.